src/VisualStudio/Core/Def/Implementation/TableDataSource/Suppression/VisualStudioDiagnosticListSuppressionStateService.cs

```diff
@@ -31,6 +34,9 @@ internal class VisualStudioDiagnosticListSuppressionStateService : IVisualStudio
         private int _selectedCompilerDiagnosticItems;
         private int _selectedNonSuppressionStateItems;
 
+        private const string SynthesizedFxCopDiagnostic = "SynthesizedFxCopDiagnostic";
+        private readonly string[] SynthesizedFxCopDiagnosticCustomTags = new string[] { SynthesizedFxCopDiagnostic };
+
         [ImportingConstructor]
         public VisualStudioDiagnosticListSuppressionStateService(
             SVsServiceProvider serviceProvider,
```

Comment:
is this service guaranteed to be created from UI thread? if so, can it derive from ForgroundAfinityObject thing.